Xamarin.Essentials/Types/Calendar.shared.cs

```diff
@@ -43,32 +47,138 @@ public CalendarEvent(string id, string calendarId, string title)
 
         public string Location { get; set; }
 
-        public bool AllDay { get; set; }
+        public bool AllDay
+        {
+            get => !EndDate.HasValue;
+            set => EndDate = value ? (DateTimeOffset?)null : StartDate;
+        }
 
         public DateTimeOffset StartDate { get; set; }
 
-        public DateTimeOffset EndDate { get; set; }
+        public DateTimeOffset? EndDate { get; set; }
+
+        public TimeSpan? Duration
+        {
+            get => EndDate.HasValue ? EndDate - StartDate : null;
+            set => EndDate = value.HasValue ? StartDate.Add(value.Value) : (DateTimeOffset?)null;
+        }
 
-        public TimeSpan Duration =>
-            AllDay ? TimeSpan.FromDays(1) : EndDate - StartDate;
+        public string Url { get; set; }
 
         public IEnumerable<CalendarEventAttendee> Attendees { get; set; }
+
+        public IEnumerable<CalendarEventReminder> Reminders { get; set; }
+
+        public RecurrenceRule RecurrancePattern { get; set; }
     }
 
+    [Preserve(AllMembers = true)]
     public class CalendarEventAttendee
     {
-        public CalendarEventAttendee()
-        {
-        }
-
-        public CalendarEventAttendee(string name, string email)
-        {
-            Name = name;
-            Email = email;
-        }
-
         public string Name { get; set; }
 
         public string Email { get; set; }
+
+        public AttendeeType Type { get; set; }
+
+        public bool IsOrganizer { get; set; }
+    }
+
+    [Preserve(AllMembers = true)]
+    public class CalendarEventReminder
+    {
+        public int MinutesPriorToEventStart { get; set; }
+    }
+
+    [Preserve(AllMembers = true)]
+    public class RecurrenceRule
+    {
+        public uint? TotalOccurrences { get; set; }
+
+        public uint Interval { get; set; }
+
+        public DateTimeOffset? EndDate { get; set; }
+
+        public RecurrenceFrequency? Frequency { get; set; }
+
+        // Only allow event to occur on these days [not available for daily]
+        public List<CalendarDayOfWeek> DaysOfTheWeek { get; set; }
```

Comment:
This is a list, and the enum is a flag.

But this is also a opportunity here... @jamesmontemagno @Redth ... Do we like/prefer enum flags vs list of classes? We can also do that Stringly typed constants. Like we do for the OS. Then we can even add methods to it. Maybe `Weekday.ToString() == "Monday"` and then `Weekday.ToShortString() == "Mon"` or something?